pkg/handler/common/provider/kubevirt.go

```diff
@@ -224,3 +228,28 @@ func KubeVirtStorageClassesWithClusterCredentialsEndpoint(ctx context.Context, u
 
 	return KubeVirtStorageClasses(ctx, kvKubeconfig)
 }
+
+func filterVMIPresets(vmiPresets apiv2.VirtualMachineInstancePresetList, quota kubermaticv1.MachineDeploymentVMResourceQuota) apiv2.VirtualMachineInstancePresetList {
+	filteredVMIPresets := apiv2.VirtualMachineInstancePresetList{}
+
+	// Range over the records and apply all the filters to each record.
+	// If the record passes all the filters, add it to the final slice.
+	for _, vmiPreset := range vmiPresets {
+		presetSpec := kubevirtv1.VirtualMachineInstancePresetSpec{}
+		if err := json.Unmarshal([]byte(vmiPreset.Spec), &presetSpec); err != nil {
+			return filteredVMIPresets
+		}
+
+		var cpu, memory int
+		if cpu = int(presetSpec.Domain.Resources.Requests.Cpu().Value()); cpu == 0 {
+			// defaults to 1 (reference: https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-config/virt-config.go#L44)
```

Comment:
1 or 100m ?
Can we directly use the `DefaultCPURequest` rather than hard code it here ?

Reply:
`FilterCPU(record, min, max int)` works with integer, so that's why used `1`.

Reply:
But if the quota is 1 to 32 for example, 100m should be rejected. Here it would be accepted.